### DISCUSSION TITLE
What is the precedence of `full_pod_spec` in `KubernetesPodOperator`?

### DISCUSSION THREAD

**Discussion by ketozhang** [apache/airflow#35425] [Q&A] [answered]
If I specify `full_pod_spec` and `image` to `KubernetesPodOperator`, is the image field taken from the the former or latter?

**Comment by Inetov** [comment, marked as answer]
as far as I understand: `full_pod_spec` is used as a base template, the values of which are overridden.
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/pod.py#L894
I think the `image` will be used

**Reply by ketozhang** [reply]
Oh! I did not expect that at all from the name. Let's get this documented 🦾

**Reply by Inetov** [reply]
> Let's get this documented

good idea, but I won't be able to do it anytime soon

**Reply by ketozhang** [reply]
I will try to get around it https://github.com/apache/airflow/issues/35509

**Comment by ketozhang** [comment]
This is now documented at https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html#argument-precedence
